ecl/hql/hqlgram2.cpp

```diff
@@ -9271,6 +9279,84 @@ bool HqlGram::checkAllowed(const attribute & errpos, const char *category, const
     return true;
 }
 
+bool HqlGram::checkDFSfields(IHqlExpression *dfsRecord, IHqlExpression *defaultRecord, const attribute& errpos)
+{
+    if (dfsRecord==defaultRecord)
+        return true;
+    bool ret = true;
+    IHqlSimpleScope *srcScope = dfsRecord->querySimpleScope();
+    ForEachChild(idx, defaultRecord)
+    {
+        IHqlExpression *subfield = defaultRecord->queryChild(idx);
+        switch (subfield->getOperator())
+        {
+        case no_record:
+        case no_ifblock:
+        case no_field:
+            {
+                OwnedHqlExpr match = srcScope->lookupSymbol(subfield->queryId());
+                if (!match)
+                {
+                    reportError(HQLERR_DFSlookupIncompatible, errpos, "Default record does not match resolved record - field %s not found", str(subfield->queryId()));
+                    ret = false;
+                }
+                else if (subfield->getOperator() != match->getOperator())
+                {
+                    reportError(HQLERR_DFSlookupIncompatible, errpos, "Default record does not match resolved record - field %s incompatible", str(subfield->queryId()));
+                    ret = false;
+                }
+                else if (subfield->queryRecord())
+                {
+                    if (match->queryRecord())
+                        ret = checkDFSfields(match->queryRecord(), match->queryRecord(), errpos) && ret;
```

Comment:
I have changed the structure to address the first comment.

Will the checkCompatible not catch the latter?